### ISSUE TITLE
Segmentation Fault when booting Server.local -- Linux, SC3 3.10.3

### ISSUE THREAD

**Issue by scacinto** [supercollider/supercollider#4671]
I'm just copying this issue from post on scsynth.org.

Building SC3 from latest source on KDE Neon (system info below). Build flags:
```
-DSC_ABLETON_LINK=OFF DSC_ED=OFF -DSC_EL=OFF -DSC_WII=OFF -DSC_VIM=OFF -DSC_IDE=ON -DNATIVE=on
```
Cannot boot the server. Any advice appreciated. Outputs are as follows:

GUI output:
```
*** Welcome to SuperCollider 3.10.3. *** For help press Ctrl-D.
SCDoc: Indexing help-files...
SCDoc: Indexed 1908 documents in 0.6 seconds
-> 12
-> localhost
Booting server 'localhost' on address 127.0.0.1:57110.
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustJPverbRaw numControls=11
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustGreyholeRaw numControls=7
Server 'localhost' exited with exit code 0.
```
Terminal output:
```
nodenoise@nodenoise-Precision-Tower-3620:~$ scsynth -u 57110
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustJPverbRaw numControls=11
Faust: supercollider.cpp: sc_api_version = 3
Faust: FaustGreyholeRaw numControls=7
Segmentation fault
```
User is in the audio group which is configured correctly, jackd has been started, etc.
Running:
```
$ scsynth -u 57110 -U "/usr/local/share/SuperCollider/Extensions/SC3plugins/"
JackDriver: client name is 'SuperCollider'
SC_AudioDriver: sample rate = 48000.000000, driver's block size = 512
SuperCollider 3 server ready.
Zeroconf: registered service 'SuperCollider'
```
works from command line...

Additionally, Server.internal works just fine:
```
s = Server.internal;
Server.default = s;
s.boot;
s.makeGui
{SinOsc.ar()}.play;
```
results in the ubiquitous sine tone we all know and love. Are there compile flags I’m using (or should) or missing dependencies that could screw up the scsynth executable but not produce errors?

Post window output from the above:

```
-> internal
changed default server to: internal
-> Server
-> internal
Booting internal server.
JackDriver: client name is 'SuperCollider'
SC_AudioDriver: sample rate = 48000.000000, driver's block size = 512
JackDriver: connected  system:capture_1 to SuperCollider:in_1
JackDriver: connected  system:capture_2 to SuperCollider:in_2
JackDriver: connected  SuperCollider:out_1 to system:playback_1
JackDriver: connected  SuperCollider:out_2 to system:playback_2
JackDriver: max output latency 32.0 ms
Requested notification messages from server 'internal'
internal: keeping clientID (0) as confirmed by server process.
Shared memory server interface initialized
-> a QWindow
-> Synth('temp__3' : 1000)
```

System Info:
Operating System: KDE neon 5.17
KDE Plasma Version: 5.17.4
KDE Frameworks Version: 5.64.0
Qt Version: 5.13.2
Kernel Version: 5.0.0-37-lowlatency
OS Type: 64-bit
Processors: 8 × Intel® Xeon® CPU E3-1240 v6 @ 3.70GHz
Memory: 31.3 GiB of RAM

**Comment by scacinto** [comment]
Problem is caused by a LADSPA Plugin "ZamTube-ladpa.so" installed as part of the KXStudio meta-plugins package. Below is the last of the output from valgrind:

```
==26886== 45000 errors in context 237 of 239:
==26886== Use of uninitialised value of size 8
==26886==    at 0x65612BB: __exp1_fma (e_exp.c:273)
==26886==    by 0x6562718: __ieee754_pow_fma (e_pow.c:115)
==26886==    by 0x20FAAAD3: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA655: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA9BA: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAC5EB: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FACF63: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FA872F: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x4010732: call_init (dl-init.c:72)
==26886==    by 0x4010732: _dl_init (dl-init.c:119)
==26886==    by 0x40151FE: dl_open_worker (dl-open.c:522)
==26886==    by 0x6BF72DE: _dl_catch_exception (dl-error-skeleton.c:196)
==26886==    by 0x40147C9: _dl_open (dl-open.c:605)
==26886== 
==26886== 
==26886== 45000 errors in context 238 of 239:
==26886== Use of uninitialised value of size 8
==26886==    at 0x65612A0: __exp1_fma (e_exp.c:274)
==26886==    by 0x6562718: __ieee754_pow_fma (e_pow.c:115)
==26886==    by 0x20FAAAD3: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA655: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA9BA: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAC5EB: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FACF63: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FA872F: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x4010732: call_init (dl-init.c:72)
==26886==    by 0x4010732: _dl_init (dl-init.c:119)
==26886==    by 0x40151FE: dl_open_worker (dl-open.c:522)
==26886==    by 0x6BF72DE: _dl_catch_exception (dl-error-skeleton.c:196)
==26886==    by 0x40147C9: _dl_open (dl-open.c:605)
==26886== 
==26886== 
==26886== 45000 errors in context 239 of 239:
==26886== Conditional jump or move depends on uninitialised value(s)
==26886==    at 0x65611F4: __exp1_fma (e_exp.c:252)
==26886==    by 0x6562718: __ieee754_pow_fma (e_pow.c:115)
==26886==    by 0x20FAAAD3: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA655: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAA9BA: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FAC5EB: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FACF63: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x20FA872F: ??? (in /usr/lib/ladspa/ZamTube-ladspa.so)
==26886==    by 0x4010732: call_init (dl-init.c:72)
==26886==    by 0x4010732: _dl_init (dl-init.c:119)
==26886==    by 0x40151FE: dl_open_worker (dl-open.c:522)
==26886==    by 0x6BF72DE: _dl_catch_exception (dl-error-skeleton.c:196)
==26886==    by 0x40147C9: _dl_open (dl-open.c:605)
==26886== 
==26886== ERROR SUMMARY: 2062836 errors from 239 contexts (suppressed: 0 from 0)
```
The plugin does not cause the error using Server.internal. Removing the plugin solves the segfault issue with Server.local. Don't know whose "fault" this is.

**Comment by jamshark70** [comment]
My opinion:

https://github.com/supercollider/sc3-plugins/issues/259

> LadspaUGen hasn't been maintained in several years. See #23 (4.5 years old, unfixed) and #210 (9/2018).
> 
> If it isn't going to be fixed (and I have no confidence that it will be), then we should provide a way for users to turn it off.

**Comment by zamaudio** [comment]
@scacinto If you are interested to take this further, can you please compile https://github.com/zamaudio/zam-plugins (master) and check if the same thing happens with ZamTube ladspa? I cannot reproduce the bug on my end with just my ladspa plugin.  (Happy to provide the latest binary with debugging symbols).

**Comment by zamaudio** [comment]
[ZamTube-ladspa.so.gz](https://github.com/supercollider/supercollider/files/3958696/ZamTube-ladspa.so.gz)

**Comment by zamaudio** [comment]
This was my bug, fixed in zamaudio/zam-plugins@87fdee6
